app/models/spree/payment_decorator.rb

```diff
@@ -0,0 +1,29 @@
+Spree::Payment.class_eval do
+
+  after_save :send_risk_notification, if: :is_risky?
+
+  def is_risky?
+    is_cvv_risky? || is_avs_risky? || state == "failed"
```

Comment:
Yes, according to [`Spree::Order#is_risky?`](https://github.com/spree/spree/blob/3-0-stable/core/app/models/spree/order.rb#L499-L501), orders riskiness is handled by [`Spree::Payment.risky`](https://github.com/spree/spree/blob/3-0-stable/core/app/models/spree/payment.rb#L59) scope which considers the `failed` state as a condition to make the payment, and order risky.